src/agent/srcview/src/report.rs

```diff
@@ -366,30 +377,35 @@ impl Report {
             .context("system time before unix epoch")?
             .as_secs();
 
-        ew.write(
-            XmlEvent::start_element("coverage")
-                .attr("lines-valid", &format!("{}", self.overall.lines))
-                .attr("lines-covered", &format!("{}", self.overall.hits))
-                .attr(
+        ew.write_event(Event::Start(
+            BytesStart::new("coverage").with_attributes([
+                ("lines-valid", format!("{}", self.overall.lines).as_str()),
+                ("lines-covered", format!("{}", self.overall.hits).as_str()),
+                (
                     "line-rate",
-                    &format!(
+                    format!(
                         "{:.02}",
                         self.overall.hits as f32 / self.overall.lines as f32
-                    ),
-                )
-                .attr("branches-valid", "0")
-                .attr("branches-covered", "0")
-                .attr("branch-rate", "0")
-                .attr("timestamp", &format!("{}", unixtime))
-                .attr("complexity", "0")
-                .attr("version", "0.1"),
-        )?;
-        ew.write(XmlEvent::start_element("sources"))?;
-        ew.write(XmlEvent::start_element("source"))?;
-        ew.write(XmlEvent::characters(""))?;
-        ew.write(XmlEvent::end_element())?; // source
-        ew.write(XmlEvent::end_element())?; // sources
-        ew.write(XmlEvent::start_element("packages"))?;
+                    )
+                    .as_str(),
+                ),
+                ("branches-valid", "0"),
+                ("branches-covered", "0"),
+                ("branch-rate", "0"),
+                ("timestamp", format!("{}", unixtime).as_str()),
+                ("complexity", "0"),
+                ("version", "0.1"),
+            ]),
+        ))?;
+
+        ew.create_element("sources").write_inner_content(|ew| {
+            ew.create_element("source")
+                .write_text_content(BytesText::new(""))?;
```

Comment:
Is it required to write an empty text node?

Reply:
I'm not sure if this is required vs. just writing an `Empty` node, though this should be the same as what the previous code was doing.

Reply:
@DrChat yes, and I don't care enough to find out 🙂